cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -131,13 +159,36 @@ public Future<KafkaStatus> createOrUpdate(Reconciliation reconciliation, Kafka k
                 status.setClusterId(kafkaAssembly.getStatus().getClusterId());
             }
 
+            if (status.getOperatorLastSuccessfulVersion() == null
+                    && kafkaAssembly.getStatus() != null
+                    && kafkaAssembly.getStatus().getOperatorLastSuccessfulVersion() != null
+            )  {
+                // If not set in the status prepared by reconciliation but set in the status previously, we copy the
+                // operatorLastSuccessfulVersion version into the new status. This is useful for example when the reconciliation fails for some
+                // reason before setting the cluster ID
+                status.setOperatorLastSuccessfulVersion(kafkaAssembly.getStatus().getOperatorLastSuccessfulVersion());
+            }
+
+
```

Comment:
Double empty line not needed?

Reply:
Apologies, I originally edited this file and didn't revert all changes. Will fix now.